bukkit/src/main/java/com/github/games647/fastlogin/bukkit/listener/protocollib/NameCheckTask.java

```diff
@@ -70,13 +69,18 @@ public NameCheckTask(FastLoginBukkit plugin, PacketEvent packetEvent, Random ran
     @Override
     public void run() {
         try {
-            // check if the player is connecting through Geyser
-            if (!plugin.getCore().getConfig().get("allowFloodgateNameConflict").toString().equalsIgnoreCase("false")
-                    && getFloodgatePlayer(username) != null) {
-                plugin.getLog().info("Skipping name conflict checking for player {}", username);
-                return;
+            ProtocolLibLoginSource source = new ProtocolLibLoginSource(packetEvent, player, random, publicKey);
+
+            //check if the player is connecting through Floodgate
+            FloodgateHook floodgateHook = new FloodgateHook(plugin);
```

Comment:
Initiating this doesn't make much sense, because it doesn't have any player specific state. It would be better to have a single instance in the protocol classes and then pass the service through to the tasks.